scaleway/table_scaleway_registry_namespace.go

```diff
@@ -0,0 +1,247 @@
+			},
+			{
+				Name:        "status",
+				Description: "The current status of the namespace.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Status").Transform(transform.ToString),
+			},
+			{
+				Name:        "status_message",
+				Description: "The current status of the namespace.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("StatusMessage").Transform(transform.ToString),
+			},
+			{
+				Name:        "endpoint",
+				Description: "The endpoint reachable by docker of the namespace.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "is_public",
+				Description: "The namespace visibility policy.",
+				Type:        proto.ColumnType_BOOL,
+				Transform:   transform.FromField("IsPublic").Transform(transform.ToBool),
+			},
+			{
+				Name:        "size",
+				Description: "total size of the namespace, calculated as the sum of the size of all images in the namespace.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Size").Transform(transform.ToString),
+			},
+			{
+				Name:        "created_at",
+				Description: "The time when the namespace was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "updated_at",
+				Description: "The time when the namespace was updated.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "image_count",
+				Description: "The number of images in the namespace.",
+				Type:        proto.ColumnType_INT,
+				Transform:   transform.FromField("ImageCount").Transform(transform.ToInt),
+			},
+			{
+				Name:        "region",
+				Description: "Specifies the region where the namespace.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Region").Transform(transform.ToString),
+			},
+
+			// Scaleway standard columns
+			{
+				Name:        "project",
+				Description: "The ID of the project where the namespace resides.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "organization",
+				Description: "The ID of the organization where the namespace resides.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: "Title of the resource.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listRegistryNamespace(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	parseRegionData, err := scw.ParseRegion(region)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_regsitry_namespace.listRegsitryNamespace", "region_parsing_error", err)
+		return nil, err
+	}
+
+	quals := d.KeyColumnQuals
+	if quals["region"] != nil && quals["region"].GetStringValue() != region {
+		return nil, nil
+	}
```

Comment:
```suggestion
	if quals["region"] == nil && quals["region"].GetStringValue() == region {
		return nil, nil
	}
```
Shouldn't it be returning nil when the quals are nil?

Reply:
maybe, i read the code. for me we don't list regsitry if we have not a region in the list and if the region on connection isn't the same.. 

i have adapted this code https://github.com/turbot/steampipe-plugin-scaleway/blob/main/scaleway/table_scaleway_instance_server.go#L225  to change zone by region.